### DISCUSSION TITLE
How can I use 'app' in #[tauri::command]

### DISCUSSION THREAD

**Discussion by zhibisora** [tauri-apps/tauri#5081] [Q&A] [answered]
How can I use 'app' in #[tauri::command]

like this

![image](https://user-images.githubusercontent.com/73344387/187120757-c74fd00d-5585-4812-a356-48624e7eae2d.png)

Maybe there is something in the 'invoke handler' that I don't know about.

**Comment by abisar** [comment, marked as answer]
I missed it too when I first started with tauri. It's here:
https://tauri.app/v1/guides/features/command/#accessing-an-apphandle-in-commands

```rust
#[tauri::command]
fn open_devtools(app_handle: AppHandle) {
    app_handle.get_window("main").unwrap().open_devtools();
}
```

**Reply by zhibisora** [reply]
Great, thanks.
